javalib/src/main/scala/java/io/OutputStreamWriter.scala

```diff
@@ -45,7 +45,11 @@ class OutputStreamWriter(private[this] var out: OutputStream,
     this(out, Charset.defaultCharset)
 
   def this(out: OutputStream, charsetName: String) =
-    this(out, Charset.forName(charsetName))
+    this(out, try {
+      Charset.forName(charsetName)
+    } catch {
+      case _: UnsupportedCharsetException => throw new UnsupportedEncodingException(charsetName)
+    })
```

Comment:
Please but [braces around multi line methods][1]

[1]: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#braces

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3687/commits/cd3cb270eb7aae92d9517de0aec034f7782a6211

Reply:
The method def still needs braces (see @sjrd's example)